frontend/simple/views/Invite.vue

```diff
@@ -80,50 +121,68 @@ export default {
     }
   },
   methods: {
-    add: async function () {
-      if (this.searchUser) {
-        if (this.searchUser === this.$store.state.loggedIn) {
-          this.self = true
-          return
-        } else {
-          this.self = false
-        }
-        try {
-          let contractId = await namespace.lookup(this.searchUser)
-          if (!this.members.find(member => member.name === this.searchUser)) {
-            this.members.push({name: this.searchUser, contractId: contractId})
-          }
-          this.searchUser = null
-          this.error = false
-        } catch (ex) {
-          this.error = true
+    async add () {
+      if (!this.searchUser) return
+
+      if (this.searchUser === this.$store.state.loggedIn) {
+        this.self = true
+        return
+      } else {
+        this.self = false
+      }
+
+      try {
+        const contractId = await namespace.lookup(this.searchUser)
+        if (!this.members.find(member => member.name === this.searchUser)) {
+          this.members.push({ name: this.searchUser, contractId })
         }
+        this.searchUser = null
+        this.error = false
+      } catch (err) {
+        this.error = true
       }
     },
-    remove: function (index) {
+    remove (index) {
       this.members.splice(index, 1)
     },
-    submit: async function () {
+    async submit () {
       try {
         this.errorMsg = null
-        // TODO: as members are successfully invited display in a seperate invitees grid and add them to some validation for duplicate invites
-        for (let i = 0; i < this.members.length; i++) {
-          let member = this.members[i]
+
+        // TODO: as members are successfully invited display in a
+        // seperate invitees grid and add them to some validation for duplicate invites
+        await this.members.map(async (member) => {
```

Comment:
This doesn't work the way you think it does. The map function does not return a promise so there is nothing for the await to wait on. To the human eye the behavior looks right but what is actually happening is it is immediately invoking the `this.invited = true` putting up a success message then running the async functions in the map. This is what is throwing off the test cases. The work isn't actually being done when the test case detects the success. It then transitions the page in the next test case aborting the prior work. If you use a for looping structure instead of map on the same level as the initial async you can make it synchornize correctly.  We are not using the results of the maps so its not needed as a loop structure